src/Couchbase.Lite.Shared/API/Database/Database.cs

```diff
@@ -179,11 +180,11 @@ public string Path
 
         [@NotNull]
         [@ItemNotNull]
-        internal ICollection<XQuery> ActiveLiveQueries { get; } = new HashSet<XQuery>();
+        internal ConcurrentDictionary<XQuery, int> ActiveLiveQueries { get; } = new ConcurrentDictionary<XQuery, int>();
```

Comment:
What is the point of making these concurrent if you wrap them all in thread safety anyway?

Reply:
My guess: To allow a non-locked loop in the close method?

Reply:
yes, I believe that's right.